geopandas/io/file.py

```diff
@@ -414,16 +421,27 @@ def infer_schema(df):
     from collections import OrderedDict
 
     # TODO: test pandas string type and boolean type once released
-    types = {"Int64": "int", "string": "str", "boolean": "bool"}
+    types = {
+        "int64": "int64",
```

Comment:
Based upon https://github.com/geopandas/geopandas/pull/2265/files#r777001174
>So my understanding currently is that when we use "int" here, fiona will automatically use this as "int64" (if your GDAL is recent enough to support int64). And only if explicitly mentioning "int32", it will use the int32 gdal type.
But so for the drivers that support int64 types, we should continue to use that by default.

I think this should be left as mapping to plain "int"
```suggestion
        "int64": "int",
```